keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,903 @@
+A claim to create Bucket. BucketClaim is namespace-scoped
 
-```
-message ProvisionerDeleteBucketRequest {
-    // This field is REQUIRED
-    string bucket_name = 1;
+```yaml
+BucketClaim {
+  TypeMeta
+  ObjectMeta
+
+  Spec BucketClaimSpec {
+    // Name of the BucketClass
+    BucketClassName string
     
-    map<string,string> bucket_context = 2;    
-}
+    // Protocols are the set of data API this bucket is required to support.
+    // The possible values for protocol are:
+    // -  S3: Indicates Amazon S3 protocol
+    // -  Azure: Indicates Microsoft Azure BlobStore protocol
+    // -  GCS: Indicates Google Cloud Storage protocol
+    Protocols []Protocol
+
+    // Name of a bucket object that was manually 
+    // created to import a bucket created outside of COSI
+    // If unspecified, then a new Bucket will be dynamically provisioned
+    // +optional
+    ExistingBucketName string
+  }
+
+  Status BucketClaimStatus {
+    // BucketReady indicates that the bucket is ready for consumpotion
+    // by workloads
+    BucketReady bool
+
+    // BucketName is the name of the provisioned Bucket in response
+    // to this BucketClaim. It is generated and set by the COSI controller 
+    // before making the creation request to the OSP backend.
+    // +optional
+    BucketName string
+  }
+```
+## BucketClass
+
+Resouce for configuring common properties for multiple Buckets. BucketClass is cluster-scoped.
 
-message ProvisionerDeleteBucketResponse {
-     // Intentionally left blank
+```yaml
+BucketClass {
+  TypeMeta
+  ObjectMeta
+
+  // DriverName is the name of driver associated with this bucket
+  DriverName string
+
+  // DeletionPolicy is used to specify how COSI should handle deletion of this
+  // bucket. There are 3 possible values:
+  //  - Retain: Indicates that the bucket should not be deleted from the OSP
+  //  - Delete: Indicates that the bucket should be deleted from the OSP
+  //        once all the workloads accessing this bucket are done
+  DeletionPolicy DeletionPolicy
+
+  // Parameters is an opaque map for passing in configuration to a driver
+  // for creating the bucket
+  // +optional
+  Parameters map[string]string
 }
 ```
 
-## ProvisionerGrantBucketAccess
+## BucketAccess
 
-This call grants access to a particular principal. The principal is the account for which this access should be granted. 
+A resource to access a Bucket. BucketAccess is namespace-scoped
 
-If the principal is set, then it should be used as the username of the created credentials or in some way should be deterministically used to generate a new credential for this request. This principal will be used as the unique identifier for deleting this access by calling ProvisionerRevokeBucketAccess
+```yaml
+BucketAccess {
+  TypeMeta
+  ObjectMeta
+
+  Spec BucketAccessSpec {
+    // BucketClaimName is the name of the BucketClaim.
+    // Exactly one of BucketClaimName or BucketName must be set.
```

Comment:
`BucketName` is mentioned here, but I don't see it in `BucketAccessSpec`. Is it on another object?

Reply:
That comment is a remnant from a previous design. Nice catch, I'll remove that line